atom/browser/ui/views/menu_bar.cc

```diff
@@ -24,26 +24,62 @@ namespace atom {
 
 namespace {
 
-const char kViewClassName[] = "ElectronMenuBar";
+#if defined(USE_X11)
+#if GTK_CHECK_VERSION(3, 0, 0)
 
-// Default color of the menu bar.
-const SkColor kDefaultColor = SkColorSetARGB(255, 233, 233, 233);
+SkColor GdkRgbaToSkColor(const GdkRGBA& rgba) {
+  return SkColorSetARGB(rgba.alpha*255, rgba.red*255,
+                        rgba.green*255, rgba.blue*255);
+}
 
-#if defined(USE_X11)
+SkColor GetStyleContextFgColor(GtkStyleContext* style_context,
+                               GtkStateFlags state) {
+  GdkRGBA rgba;
+  gtk_style_context_get_color(style_context, state, &rgba);
+  return GdkRgbaToSkColor(rgba);
+}
+
+SkColor GetStyleContextBgColor(GtkStyleContext* style_context,
+                               GtkStateFlags state) {
+  GdkRGBA rgba;
+  gtk_style_context_get_background_color(style_context, state, &rgba);
+  return GdkRgbaToSkColor(rgba);
+}
+
+// GTK+ 3 impl
 void GetMenuBarColor(SkColor* enabled, SkColor* disabled, SkColor* highlight,
                      SkColor* hover, SkColor* background) {
   GtkWidget* menu_bar = gtk_menu_bar_new();
+  GtkStyleContext* sc = gtk_widget_get_style_context(menu_bar);
+  *enabled    = GetStyleContextFgColor(sc, GTK_STATE_FLAG_NORMAL);
+  *disabled   = GetStyleContextFgColor(sc, GTK_STATE_FLAG_INSENSITIVE);
+  *highlight  = GetStyleContextFgColor(sc, GTK_STATE_FLAG_SELECTED);
+  *hover      = GetStyleContextFgColor(sc, GTK_STATE_FLAG_PRELIGHT);
+  *background = GetStyleContextBgColor(sc, GTK_STATE_FLAG_NORMAL);
+  g_object_unref(G_OBJECT(menu_bar));
+}
 
+#else  // GTK_CHECK_VERSION(3,0,0)
+
+// GTK+ 2 impl
+void GetMenuBarColor(SkColor* enabled, SkColor* disabled, SkColor* highlight,
+                     SkColor* hover, SkColor* background) {
+  GtkWidget* menu_bar = gtk_menu_bar_new();
   GtkStyle* style = gtk_rc_get_style(menu_bar);
-  *enabled = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_NORMAL]);
-  *disabled = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_INSENSITIVE]);
-  *highlight = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_SELECTED]);
-  *hover = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_PRELIGHT]);
+  *enabled    = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_NORMAL]);
+  *disabled   = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_INSENSITIVE]);
+  *highlight  = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_SELECTED]);
+  *hover      = libgtkui::GdkColorToSkColor(style->fg[GTK_STATE_PRELIGHT]);
```

Comment:
Is linter ok with this?

Reply:
Seems so; all 8 checks passed :smile:

Reply:
That's weird =/ Ok.

Reply:
Do you use clang-format? `git-clang-format efb44050dbaab32128777cf70c0095d43f13083e` changes a few things in this branch. It would be nice if everyone used clang-format, I'm not sure what the policy is right now.